### PR TITLE
Reorganize repo

### PR DESCRIPTION
**Pull request by raimis**
This is preparation for a `conda-forge` package.

- [x] Reorganize the files
- [x] Improve the build system
  - [x] Build the library
  - [x] Build the test executables
  - [x] Run the tests
- [x] Update the documentation

### CODE REVIEW

File: src/schnet/CudaCFConv.cu [hunk 1]
```diff
@@ -260,11 +260,11 @@ const float* CudaCFConv::ensureOnDevice(const float* arg, float*& deviceMemory,
     return (const float*) attrib.devicePointer;
 }
 
-__device__ float cutoffFunction(float r, float rc) {
+__device__ float cutoffFunction2(float r, float rc) {
```
Comment (peastman) [inline src/schnet/CudaCFConv.cu:263]: Why renaming these functions?

Comment (raimis) [reply src/schnet/CudaCFConv.cu:263]: Name collision when linking due to functions with the same names in `CudaANISymmetryFunctions.cu`

Comment (raimis) [reply src/schnet/CudaCFConv.cu:263]: Open to suggestions for a better naming scheme.

Comment (peastman) [reply src/schnet/CudaCFConv.cu:263]: How about adding the `static` keyword so it knows those functions are local to this file and shouldn't be exported?

Comment (raimis) [reply src/schnet/CudaCFConv.cu]: Fixed!

**Comment by raimis** [comment]
The minimal reorganisation towards a `conda-forge`  package is done.

@mikemhenry could you check if something more is need for a feedstock?

@peastman could you review the rest?

**Comment by mikemhenry** [comment]
This looks good! It shouldn't be too hard to package since all the dependencies are on conda-forge.

**Comment by raimis** [comment]
@peastman can we merge this?

**Comment by peastman** [comment]
Looks good.  Thanks!

**Pull request merged**